libraries/AP_Param/AP_Param.cpp

```diff
@@ -68,6 +68,18 @@ const AP_Param::Info *AP_Param::_var_info;
 struct AP_Param::param_override *AP_Param::param_overrides = nullptr;
 uint16_t AP_Param::num_param_overrides = 0;
 
+/*
+  this holds default parameters in the normal NAME=value form for a
+  parameter file. It can be manipulated by apj_tool.py to change the
+  defaults on a binary without recompiling
+ */
```

Comment:
I'm thinking that an alternative could be to let the the default values to be manipulated in the ELF binary by using special sections to hold the parameters.  This way we can have symbols around the start and end of that section and let them be manipulated with no code added to load the parameters.

Some added benefits:  we could hold default parameters for more than one vehicle (relevant for Linux boards only) in different sections and choose which one to use in runtime.

Con: we would need to pull the parameters out from their objects (e.g. they are currently inside Copter/Plane/AP_InertialSensor/etc) in order to be able to store them in a separate section.

Reply:
I think that is way too complex. Default values aren't just in the parameter tables, they are also in code and passed to constructors.
The method I've proposed works for all parameters and allows you to override just the parameter changes you need for a vehicle. We could have multiple tables on Linux and a command line option to choose, but I'd prefer to keep it simple to start with. 
Another thing we could have on Linux is a --defaults command line option that points at a file. The AP_Param code still have the method of loading from a file. We use that in SITL and there is no reason we can't do that same for Linux.

Reply:
I thought separating the parameters in an ELF section would actually be simpler, but I get your point of this being similar to what we already have (but in a "binary" format rather than text). Fair enough.